packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -0,0 +1,124 @@
+import BN from 'bn.js'
+import { takeLatest } from 'redux-saga/effects'
+import { call, put, race, select, take } from 'typed-redux-saga'
+
+import { ID } from 'models/Identifiers'
+import { isPremiumContentUSDCPurchaseGated } from 'models/Track'
+import { purchaseContent } from 'services/audius-backend/solana'
+import { getUSDCUserBank } from 'store/buy-usdc/sagas'
+import {
+  buyUSDCFlowFailed,
+  buyUSDCFlowSucceeded,
+  onRampCanceled,
+  startBuyUSDCFlow
+} from 'store/buy-usdc/slice'
+import { OnRampProvider } from 'store/buy-usdc/types'
+import { getTrack } from 'store/cache/tracks/selectors'
+import { getUser } from 'store/cache/users/selectors'
+import { getContext } from 'store/effects'
+
+import { startPurchaseContentFlow } from './slice'
+import { ContentType } from './types'
+
+type GetPurchaseConfigArgs = {
+  contentId: ID
+  contentType: ContentType
+}
+
+function* getPurchaseConfig({ contentId, contentType }: GetPurchaseConfigArgs) {
+  if (contentType !== ContentType.TRACK) {
+    throw new Error('Only tracks are supported')
```

Comment:
either the caller or this method internally should probably handle the exceptions and gracefully inform the user

Reply:
Oh yes. I believe I intended to wrap a try/catch around this in the calling function and propagate the error up.